src/integration/test_suites/smoke_test_fc_ratchet_seed_write_read/smoke_test_fc_ratchet_seed_write_read.c

```diff
@@ -0,0 +1,180 @@
+//********************************************************************************
+// SPDX-License-Identifier: Apache-2.0
+//
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//********************************************************************************
+#include <string.h>
+#include <stdint.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include "soc_address_map.h"
+#include "printf.h"
+#include "riscv_hw_if.h"
+#include "soc_ifc.h"
+#include "caliptra_ss_lc_ctrl_address_map.h"
+#include "caliptra_ss_lib.h"
+#include "fuse_ctrl.h"
+#include "lc_ctrl.h"
+#include "fuse_ctrl_mmap.h"
+
+volatile char* stdout = (char *)SOC_MCI_TOP_MCI_REG_DEBUG_OUT;
+#ifdef CPT_VERBOSITY
+    enum printf_verbosity verbosity_g = CPT_VERBOSITY;
+#else
+    enum printf_verbosity verbosity_g = LOW;
+#endif
+
+
+// Writes the entire partition with values wdata0 and wdata1 and reads back the values written match the values read
+uint32_t write_read_entire_partition(uint32_t num_partition, uint32_t wdata0, uint32_t wdata1, uint32_t write_error_mask, uint32_t read_error_mask) {
+  uint32_t rdata[2];
+
+  for (uint32_t addr = partitions[num_partition].address; addr < partitions[num_partition].digest_address; addr += partitions[num_partition].granularity / 8) {
+    dai_wr(addr, wdata0, wdata1, partitions[num_partition].granularity, write_error_mask);
+    dai_rd(addr, &rdata[0], &rdata[1], partitions[num_partition].granularity, read_error_mask);
+    if (rdata[0] != wdata0 || (partitions[num_partition].granularity > 32 && rdata[1] != wdata1)) {
+      VPRINTF(LOW, "ERROR: partitions[num_partition=%0d] read_value, does not match written_value\n", num_partition);
+      return 0;
+    }
+  }
+  return 1;
+
+}
+// Zeroizes the fuse, marker and digest
+uint32_t zeroize_all_partition(uint32_t num_partition) {
+  uint32_t r_data[2];
+
+
+  memset(r_data, 0, 2*sizeof(uint32_t));
```

Comment:
```suggestion
  memset(r_data, 0, sizeof(r_data));
  ```
  The same for the other `memset` bellow.